### ISSUE TITLE
Please Reclassify Error: "a function .... must return a value" ts(2355) as a Warning

### ISSUE THREAD

**Issue by bf-software** [microsoft/TypeScript#60430]
### 🔍 Search Terms

ts(2355)
A function whose declared type is neither 'undefined', 'void', nor 'any' must return a value.

### ✅ Viability Checklist

- [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
- [x] This wouldn't change the runtime behavior of existing JavaScript code
- [x] This could be implemented without emitting different JS based on the types of the expressions
- [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, new syntax sugar for JS, etc.)
- [x] This isn't a request to add a new utility type: https://github.com/microsoft/TypeScript/wiki/No-New-Utility-Types
- [x] This feature would agree with the rest of our Design Goals: https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals

### ⭐ Suggestion

I'd like to demote error "A function whose declared type is neither 'undefined', 'void', nor 'any' must return a value. ts(2355)" to a warning.
(I couldn't figure out a way to do this in VSCode's config just for myself.)

### 📃 Motivating Example

The reason is because of this annoyance:  Every time I'm writing a function and writing out the return type, VSCode's error checker always reports an error via the red squiggly underline.  I never know if the error is a real problem or not until I hover over it with the mouse.  That is, I'm never sure if I've mistyped the type, or the type is correct, but not in scope, or more often than not, it's just the pesky ts(2355) error.  At this point, the ts(2355) error is of no use because I haven't even gotten to start writing the body of the function yet, and showing this as an error just interrupts my flow.

### 💻 Use Cases

1. What do you want to use this for?    A: smoother development flow.
2. What shortcomings exist with current approaches?    A: always reporting an obvious error & hiding possible real errors.
3. What workarounds are you using in the meantime?   A: always having to check what the error actually is, or ignoring it.

**Comment by MartinJohns** [comment]
TypeScript does not have the concept of warnings.

**Comment by RyanCavanaugh** [comment]
I understand the scenario you're in, but it's unclear how changing this to a warning changes the UX in any appreciable way. If "warnings" existed (they don't), presumably many things would be a warning, but then you'd be right back to where you started (I don't know if this warning is because I haven't returned anything yet, or for some other reason). It's not like you would want to *never* see warnings, since a function that failed to return its declared type is absolutely a real problem if you walk away from it.

**Comment by bf-software** [comment]
Ahh, got it -- thanks for the replies guys.  So if there is any remedy to this, it would have to be implemented 100% in VSCode.  (Like through some kind of ability to change the error underline color based on specific errors.)
